### DISCUSSION TITLE
If I transfer learning, do I need to freeze the model ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#1781] [General Q&A]
Hi,

I am trying and using YourTts for fun. I have understood transfer learning could help me get a French voice faster and of better quality.

That's why I am planning to transfer learning from the English (or German) VITS model to French with a homemade dataset. 
I read here (https://keras.io/guides/transfer_learning/) that it was recommended to freeze the source model. But in YourTTS doc about tranfer learning, it only mentions to use the flag resume-path. 

So my question is whether I need to set a freeze option in the config file or it is not recommended for VITS ? And if so which option(s) should be set as there seem to be 2 of them (freeze dp and freeze encoder) ? 

Moreover if model is frozen I don't see clearly what will improve in the model during training with a French dataset. 

Thanks in advance for shedding lights on this topic !

**Comment by erogol** [comment]
It depends on data, requirements, and expectations. If you want to just change the voice then freeze everything except decoder and vocoder. If you have a new accent or different style then freeze decoder (maybe). So it is hard to give a precise recipe.

**Reply by Ca-ressemble-a-du-fake** [reply]
Thanks @erogol. So let me give you more details on the method I am planning to use. 

I will start with an already existing English model that is available on this website. Then I will try to clone my voice in French with a 1h dataset or (if my recording setup proves to be insufficient) I will use a much longer dataset from an Audiobook (following the method described by Ben Andrew https://m.youtube.com/watch?v=oS5VMxbhREE). Audiobook duration is approximately 10 hours long. 

So accent, language and voice will all be different. Is it worth freezing something ?

Then when I have a French model, I will try cloning other voices with dataset durations varying between some minutes to 1 hour. Voices and accents will be different. If I got you right I should freeze the decoder. 

But I don't see an option in the config file to only freeze the vocoder (I see freeze dp and freeze encoder), is there another option to use ?

Thanks again for your help
